src/utils/bot_factory.py

```diff
@@ -35,12 +35,14 @@ def start(self):
                     # because we're in utils here we need to up a directory so to load the janitor cog from
                     # src/cogs/janitor/janitor.py, it needs to look like ..cogs.janitor.janitor
                     extension_prefix = "..cogs" if cogs_directory else f"..cogs.{directory}"
-                    initial_extensions.append(f"{extension_prefix}.{filename[:-3]}")
+                    if directory == filename[:-3]:
+                        initial_extensions.append(f"{extension_prefix}.{filename[:-3]}")
+
+        print(initial_extensions)
 
         # Here we load our extensions(cogs) listed above in [initial_extensions].
         for extension in initial_extensions:
             asyncio.run(self.bot.load_extension(name=extension, package=__package__))
 
```

Comment:
So I don't think this is going to work with how the utils cog folder is organized. Right now it has a sync cog. I in general like this approach though, maybe we can also add or filename.contains('_cog') or something, so then you just need to apend _cog to files that you need to be loaded as well

Reply:
sure, we can do something like that, ill update here shortly